File: Domain/Entities/Topic.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Domain.Entities
{
    public class Topic
    {
        public Guid Id { get; set; }
        
        [Required]
        public string Title { get; set; }
        [Required]
        public string Body { get; set; }
        [Required]
        public DateTime CreatedAt { get; set; }
        
        public Category Category { get; set; }
        
        public AppUser Creator { get; set; }
        
        public ICollection<Comment> Comments { get; set; } = new List<Comment>();
        public ICollection<Like> Likes { get; set; } = new List<Like>();
    }
}

File: forum-client/src/app/models/topic.ts
import Category from "./category";
import { Profile } from "./profile";

export interface Topic{
    id: string;
    title: string;
    createdAt: Date | null;
    body: string;
    likes: number;
    isLiked: boolean;
    category: Category;
    creator: Profile;
}

export class TopicFormValues{
    id!: string;
    title!: string ;
    createdAt!: Date | null;
    body!: string;
    likes!: number;
    isLiked!: boolean;
    category!: Category;


    constructor(topic?: Topic){
        if(topic){
           this.id = topic.id;
           this.title = topic.title;
           this.body = topic.body;
           this.likes = topic.likes;
           this.category = topic.category;
           this.isLiked = topic.isLiked
        }
    }
}
